src/lib.rs

```diff
@@ -187,6 +211,37 @@ impl Led {
         [&mut self.first, &mut self.second, &mut self.third]
     }
 
+    pub fn select_by_color(&mut self, color: LedColor) -> &mut Pin {
+        let pin: &mut Pin = match color {
+            LedColor::Green => &mut self.first,
+            LedColor::Blue => &mut self.second,
+            LedColor::Red => &mut self.third,
+        };
+        pin
+    }
+
+    pub fn set_led(&mut self, color: LedColor, state: bool) {
+        let state_u8 = if state { 0 } else { 1 }; // LEDs are enabled with logic low
```

Comment:
this is unnecessary, you can do `state.into()`

Reply:
idk why it was not working before.